docs/source/python/dataset.rst

```diff
@@ -0,0 +1,354 @@
+
+    # writing it into two parquet files
+    import pyarrow.parquet as pq
+    pq.write_table(table.slice(0, 5), base / "parquet_dataset/data1.parquet")
+    pq.write_table(table.slice(5, 10), base / "parquet_dataset/data2.parquet")
+
+Dataset discovery
+~~~~~~~~~~~~~~~~~
+
+A :class:`Dataset` object can be created with the :func:`dataset` function. We
+can pass it the path to the directory containing the data files:
+
+.. ipython:: python
+
+    dataset = ds.dataset(base / "parquet_dataset", format="parquet")
+    dataset
+
+Alternatively, it also accepts a path to a single file, or a list of file
+paths.
+
+Creating this :class:`Dataset` object did yet materialize the full dataset, but
+it crawled the directory to find all the files:
+
+.. ipython:: python
+
+    dataset.files
+
+and it did infer the schema (by default from the first file):
+
+.. ipython:: python
+
+    print(dataset.schema.to_string(show_field_metadata=False))
+
+With the :meth:`Dataset.to_table` method, we can read the dataset into a
+pyarrow Table (note this will read everything at once, which can require a lot
+of memory, see below on filtering / iterative loading):
+
+
+.. ipython:: python
+
+    dataset.to_table()
+    # converting to pandas to see the contents of the scanned table
+    dataset.to_table().to_pandas()
+
+Reading different file formats
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The above examples use Parquet files as dataset source, but the Dataset API
+aims to provide a consistent interface for multiple file formats and sources.
+Currently, Parquet and Feather / Arrow IPC file format are supported; more
+formats are planned in the future.
+
+If we save the table as a Feather file instead of Parquet files:
+
+.. ipython:: python
+
+    import pyarrow.feather as feather
+
+    feather.write_feather(table, base / "data.feather")
+
+then we can read the Feather file using the same functions, but with specifying
+``format="ipc"``:
+
+.. ipython:: python
+
+    dataset = ds.dataset(base / "data.feather", format="ipc")
+    dataset.to_table().to_pandas().head()
+
+The format name as a string, like:
+
+.. code-block::
+
+    ds.dataset(..., format="parquet")
+
+is a short-hand for
+
+.. code-block::
+
+    ds.dataset(..., format=ds.ParquetFileFormat())
+
+Those file format objects can take keywords to customize reading (see for
+example :class:`ParquetFileFormat`).
+
+
+Filtering data
+--------------
+
+To avoid reading all data when only needing a subset, the ``columns`` and
+``filter`` keywords can be used.
+
+The ``columns`` keyword can be used to only read a selection of the columns,
+and accepts a list of column names:
+
+.. ipython:: python
+
+    dataset = ds.dataset(base / "parquet_dataset", format="parquet")
+    dataset.to_table(columns=['a', 'b']).to_pandas()
+
+With the ``filter`` keyword, rows which do not match the filter predicate will
+not be included in the returned table. The keyword expects a boolean expression
```

Comment:
```suggestion
not be included in the returned table. The keyword expects a class:`Expression`
```

Reply:
You find the "boolean" wrong here?

Reply:
It's not wrong, just less specific than referencing the class `Expression` explicitly. For example a reader might imagine that `filter=True` or even `filter=lambda a: a % 2 == 0` were legal

Reply:
OK, understand now, made it "boolean :class:'Expression'" then